packages/runtime/src/types-input.ts

```diff
@@ -20,6 +21,11 @@ export const ASADefSchema = z.object({
   clawback: AddressSchema.optional(),
   optInAccNames: z.array(z.string()).optional()
 })
+  .refine(o => (totalRegex.test(String(o.total)) &&
+  BigInt(o.total) <= 0xFFFFFFFFFFFFFFFFn && BigInt(o.total) > 0n), {
+    message: "Total must be a positive number and smaller than 2^64-1 ",
```

Comment:
will this be `smaller than 2^64` ?

Reply:
[https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61](https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61)

Reply:
`assetTotal <= 0xFFFFFFFFFFFFFFFFn`
It has equal though. 0x...n this is equal to 2^64 - 1 right ?

Reply:
Right. So i think the message should be `Total must be a positive number and smaller than or equal to 2^64-1`. And the message here seems incorrect then - [https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61](https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61)

Reply:
yes or `smaller than 2^64`